drivers/include/periph/uart.h

```diff
@@ -72,95 +87,83 @@ typedef void(*uart_rx_cb_t)(void *arg, char data);
  *
  * The UART device will be initialized with the following configuration:
  * - 8 data bits
  * - no parity
  * - 1 stop bit
- * - baud-rate as given
+ * - baudrate as given
  *
- * @param[in] uart          the UART device to initialize
- * @param[in] baudrate      the desired baud-rate in baud/s
- * @param[in] rx_cb         receive callback is called for every byte the is receive
- *                          in interrupt context
- * @param[in] tx_cb         transmit callback is called when done with sending a byte
- *                          (TX buffer gets empty)
- * @param[in] arg           optional argument passed to the callback functions
+ * @param[in] uart          UART device to initialize
+ * @param[in] baudrate      desired baudrate in baud/s
+ * @param[in] rx_cb         receive callback, executed in interrupt context once
+ *                          for every byte that is received (RX buffer filled)
+ * @param[in] tx_cb         transmit callback, executed in interrupt context
+ *                          for each finished byte transfer (TX buffer empty)
+ * @param[in] arg           optional context passed to the callback functions
  *
  * @return                  0 on success
- * @return                  -1 for invalid baud-rate
- * @return                  -2 for all other errors
+ * @return                  -1 on invalid UART device
+ * @return                  -2 on inapplicable baudrate
+ * @return                  -3 on other errors
  */
-int uart_init(uart_t uart, uint32_t baudrate, uart_rx_cb_t rx_cb, uart_tx_cb_t tx_cb, void *arg);
+int uart_init(uart_t uart, uint32_t baudrate,
+              uart_rx_cb_t rx_cb, uart_tx_cb_t tx_cb, void *arg);
 
 /**
- * @brief Initialize an UART device for (conventional) blocking usage
+ * @brief   Begin a new transmission
  *
- * This function initializes the an UART device for usage without interrupts.
- * When initializing with this function, the corresponding read_blocking and
- * write_blocking functions must be used.
- *
- * The blocking mode should only be used for debugging and testing.
- *
- * Same as uart_init(), the UART device is configured with in 8N1 mode with the given baud-rate.
- *
- * @param[in] uart          the UART device to initialize
- * @param[in] baudrate      the desired baud-rate in baud/s
- *
-* @return                   0 on success, -1 for invalid baud-rate, -2 for all other errors
- */
-int uart_init_blocking(uart_t uart, uint32_t baudrate);
-
-/**
- * @brief Begin a new transmission, on most platforms this function will enable the TX interrupt
+ * When sending buffered data, we need to tell the UART device, whenever there
+ * is new data waiting to be transferred. In that case, this function should be
+ * called. On most platforms this function will simply enable the TX interrupt
+ * leading to a call of the TX callback function.
  *
  * @param[in] uart          UART device that will start a transmission
  */
 void uart_tx_begin(uart_t uart);
```

Comment:
yes, it is needed. The typical application is

- implement the TX callback so that it takes a byte out of a buffer and calls `uart_write` with that byte
- put some data in a ringbuffer
- call `uart_tx_begin`, which in turn will trigger the TX callback which calls then `uart_write`